kindedj/src/main/java/io/kindedj/Hk.java

```diff
@@ -16,5 +16,14 @@
 
 package io.kindedj;
 
-public interface HK<F, A> {
+/**
+ * Hk, or higher kind, represents a value that wraps another value. It would be equivalent to F&lt;A&gt; if Java supported higher kinds.
+ * <br/><br/>
+ * To construct higher kinds of arity 2+ you have to nest Hk into its witness type parameter.
+ * For example, to represent F&lt;G, H, A&gt; you have to create an interface or type alias Hk&lt;Hk&lt;Hk&lt;F, G&gt;, H&gt;, A&gt;
+ *
+ * @param <F> The witness type marking the wrapper type
```

Comment:
*wrapper* is a bit too much collection oriented. what about:
"
`<F>` the 'witness' type of the parametrized class/interface to be lifted as type constructor.
`<A>` actual type parameter of the lifted class/interface.
"